pulser-core/pulser/channels/eom.py

```diff
@@ -79,7 +79,15 @@ def _to_dict(self) -> dict[str, Any]:
         return obj_to_dict(self, **params)
 
     def _to_abstract_repr(self) -> dict[str, Any]:
-        return {f.name: getattr(self, f.name) for f in fields(self)}
+        all_fields = fields(self)
+        params = {}
+        defaults = get_dataclass_defaults(all_fields)
+        for f in all_fields:
+            value = getattr(self, f.name)
+            if f.name in defaults and value == defaults[f.name]:
+                continue
+            params[f.name] = value
+        return params
```

Comment:
If I get it right: When serializing an eom_config, the parameters provided are the one that have been changed.
It looks good to me, why don't we do this for channel ? (we do this only for optional parameters)

Reply:
Edit: It's because the only field in default is the optional one... Althought not necessary, won't it be better yo have the same structure as in channel, and have optional_keys here as well ?

Reply:
Edit Edit: I think it would fit well with having a variable `OPTIONAL_EOM_KEYS` shared by `eom.py` and `deserializer.py`